library/connect.c

```diff
@@ -1063,19 +1016,19 @@ static int ziti_channel_start_connection(struct ziti_conn *conn, ziti_channel_t
         init_key_pair(&conn->key_pair);
         nheaders++;
     }
-    if (req->dial_opts != NULL) {
-        if (req->dial_opts->identity != NULL) {
-            headers[nheaders].header_id = TerminatorIdentityHeader;
-            headers[nheaders].value = (uint8_t *) req->dial_opts->identity;
-            headers[nheaders].length = strlen(req->dial_opts->identity);
-            nheaders++;
-        }
-        if (req->dial_opts->app_data != NULL) {
-            headers[nheaders].header_id = AppDataHeader;
-            headers[nheaders].value = req->dial_opts->app_data;
-            headers[nheaders].length = req->dial_opts->app_data_sz;
-            nheaders++;
-        }
+
+    if (req->dial_opts.identity != NULL) {
```

Comment:
Might be worth checking for empty string here too. I see a lot of configs with dial / listen options set to `""`.

Reply:
I added the check in clone_dial_opts